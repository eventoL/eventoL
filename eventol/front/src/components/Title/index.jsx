import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

const Title = ({label, children}) => (
  <div className="row">
    <div className="col-sm-12">
      <div className="title">
        <h1>{label}</h1>
        <div className="buttons pull-right">{children}</div>
      </div>
    </div>
  </div>
);

Title.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  label: PropTypes.string,
};

Title.defaultProps = {
  children: null,
  label: '',
};

export default Title;
