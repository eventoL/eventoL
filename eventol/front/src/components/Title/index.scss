.title {
  text-align: left;

  .buttons {
    margin-top: -60px;
  }
}
