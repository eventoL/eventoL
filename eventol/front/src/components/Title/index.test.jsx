import React from 'react';
import renderer from 'react-test-renderer';

import Title from '.';

describe('Title', () => {
  let component;
  let label;
  let tree;

  const getComponent = () => {
    component = renderer.create(<Title label={label} />);
    return component;
  };

  beforeEach(() => {
    label = undefined;
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('With data', () => {
    test('Snapshot', () => {
      label = 'label';
      component = getComponent();
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
