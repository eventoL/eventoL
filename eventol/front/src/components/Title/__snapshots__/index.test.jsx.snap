// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Default Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <div
      className="title"
    >
      <h1>
        
      </h1>
      <div
        className="buttons pull-right"
      />
    </div>
  </div>
</div>
`;

exports[`Title With data Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-12"
  >
    <div
      className="title"
    >
      <h1>
        label
      </h1>
      <div
        className="buttons pull-right"
      />
    </div>
  </div>
</div>
`;
